test/src/csg/TestCSGAxialSurfaceMeshGenerator.C

```diff
@@ -0,0 +1,71 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "TestCSGAxialSurfaceMeshGenerator.h"
+#include "MeshGenerator.h"
+
+registerMooseObject("MooseTestApp", TestCSGAxialSurfaceMeshGenerator);
+
+InputParameters
+TestCSGAxialSurfaceMeshGenerator::validParams()
+{
+  InputParameters params = MeshGenerator::validParams();
+
+  params.addRequiredParam<MeshGeneratorName>("input", "The input MeshGenerator.");
+  params.addRequiredParam<Real>("axial_height", "Axial height of output CSG mesh.");
+  // Declare that this generator has a generateData method
+  MeshGenerator::setHasGenerateData(params);
+  // Declare that this generator has a generateCSG method
+  MeshGenerator::setHasGenerateCSG(params);
+  return params;
+}
+
+TestCSGAxialSurfaceMeshGenerator::TestCSGAxialSurfaceMeshGenerator(const InputParameters & params)
+  : MeshGenerator(params),
+    _mesh_ptr(getMesh("input")),
+    _axial_height(getParam<Real>("axial_height"))
+{
+}
+
+std::unique_ptr<MeshBase>
+TestCSGAxialSurfaceMeshGenerator::generate()
+{
+  auto null_mesh = nullptr;
+  return null_mesh;
+}
+
+std::unique_ptr<CSG::CSGBase>
+TestCSGAxialSurfaceMeshGenerator::generateCSG()
+{
+  std::unique_ptr<CSG::CSGBase> input_mesh = std::move(getCSGBase("input"));
+
+  auto root_univ = input_mesh->getRootUniverse();
+  const auto cell_name = "square_cell";
+  auto cell_ptr = root_univ->getCell(cell_name);
+  auto cell_region = cell_ptr->getRegion();
+  const auto centroid = Point(0, 0, 0);
+
+  // Add surfaces and halfspaces corresponding to top and bottom axial planes
+  std::vector<std::string> surf_names{"plus_z", "minus_z"};
+  std::vector<Real> coeffs{0.5 * _axial_height, -0.5 * _axial_height};
+  for (unsigned int i = 0; i < coeffs.size(); ++i)
+  {
+    const auto surf_name = "surf_" + surf_names[i];
+    // z plane equation: 0.0*x + 0.0*y + 1.0*z = (+/-)0.5 * axial_height
+    auto plane_ptr = input_mesh->createPlaneFromCoefficients(surf_name, 0.0, 0.0, 1.0, coeffs[i]);
+    const auto region_direction = plane_ptr->directionFromPoint(centroid);
+    auto halfspace =
+        ((region_direction == CSG::CSGSurface::Direction::POSITIVE) ? +plane_ptr : -plane_ptr);
+    cell_region &= halfspace;
+  }
+
+  input_mesh->updateCellRegion(cell_ptr, cell_region);
```

Comment:
why do we need to call this? could we get a reference to the cell region above instead?

Reply:
Partly this line is here to explicitly test this function. The other reason is that this is a more defensive approach to make sure that regions are updated in a way that is expected and controlled at the CSGBase level. In `updateCellRegion` we do a call to `checkRegionSurfaces` which ensures that all surfaces being applied to the region do in fact exist in the CSGBase instance. If that check weren't there or if this were to be updated on the cell/region directly, there would be no way to know if the surfaces being applied are actually a part of the CSGBase. CSGRegion and CSGCell don't have information about what surfaces exist in the instance, they just care that they receive a surface.

Reply:
Checking in @GiudGiud on this question you had. Does this clarify the need for this line? If so, can you resolve or let me know if we need to make more changes.

Reply:
yup good for me. Thank you for the explanation

Reply:
so moving forward we ll have to be careful to not forget these calls that seem like we could skip. If it turns into a recurring mistake we might want to enforce it programmatically. For now all good